.github/workflows/mobile-android_build.yml

```diff
@@ -275,7 +275,6 @@ jobs:
         ./bazelw build \
             --config=mobile-remote-release-clang \
             --fat_apk_cpu=x86_64 \
-            --define envoy_mobile_listener=enabled \
```

Comment:
as a general point - when i see the need to do this it makes me think this _should_ be defined in a bazel config

Reply:
(but that is not necessarily correct)

Reply:
So there's places where it's set true and false. 
I'm 90% sure we need it for at least one build, and I would like to fix tests so it'll run either way.  using WIP to see which builds / tests fail since I can't run them all locally